### ISSUE TITLE
Text Classification Console Codegen - DllNotFoundException: Unable to load DLL 'LibTorchSharp' 

### ISSUE THREAD

**Issue by luisquintanilla** [dotnet/machinelearning-modelbuilder#2344]
**System Information (please complete the following information):**
- Model Builder Version (available in Manage Extensions dialog): 16.14.0.2255301
- Visual Studio Version: 17.3.6

**Describe the bug**
- On which step of the process did you run into an issue: Consume
- Clear description of the problem: Error thrown because reference to TorchSharp-cpu or TorchSharp-cuda-windows missing from generated Console App project. Running the app also is blocked by #2341 

**To Reproduce**
Train text classification model using local GPU and add Console App project to solution.

**Expected behavior**
Console app runs successfully. Running 

**Screenshots**

**Additional context**

```
System.Reflection.TargetInvocationException
  HResult=0x80131604
  Message=Exception has been thrown by the target of an invocation.
  Source=System.Private.CoreLib
  StackTrace:
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Span`1& arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Microsoft.ML.Runtime.ComponentCatalog.LoadableClassInfo.CreateInstanceCore(Object[] ctorArgs)
   at Microsoft.ML.Runtime.ComponentCatalog.TryCreateInstance[TRes](IHostEnvironment env, Type signatureType, TRes& result, String name, String options, Object[] extra)
   at Microsoft.ML.Runtime.ComponentCatalog.TryCreateInstance[TRes,TSig](IHostEnvironment env, TRes& result, String name, String options, Object[] extra)
   at Microsoft.ML.ModelLoadContext.TryLoadModelCore[TRes,TSig](IHostEnvironment env, TRes& result, Object[] extra)
   at Microsoft.ML.ModelLoadContext.TryLoadModel[TRes,TSig](IHostEnvironment env, TRes& result, RepositoryReader rep, Entry ent, String dir, Object[] extra)
   at Microsoft.ML.ModelLoadContext.LoadModel[TRes,TSig](IHostEnvironment env, TRes& result, RepositoryReader rep, Entry ent, String dir, Object[] extra)
   at Microsoft.ML.ModelLoadContext.LoadModelOrNull[TRes,TSig](IHostEnvironment env, TRes& result, RepositoryReader rep, String dir, Object[] extra)
   at Microsoft.ML.ModelLoadContext.LoadModel[TRes,TSig](IHostEnvironment env, TRes& result, RepositoryReader rep, String dir, Object[] extra)
   at Microsoft.ML.ModelOperationsCatalog.Load(Stream stream, DataViewSchema& inputSchema)
   at Microsoft.ML.ModelOperationsCatalog.Load(String filePath, DataViewSchema& inputSchema)
   at TextClassification_ConsoleApp1.TextClassification.CreatePredictEngine() in C:\Dev\MBBugBash112022\TextClassification_ConsoleApp1\TextClassification.consumption.cs:line 70
   at TextClassification_ConsoleApp1.TextClassification.<>c.<.cctor>b__9_0() in C:\Dev\MBBugBash112022\TextClassification_ConsoleApp1\TextClassification.consumption.cs:line 54
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)

  This exception was originally thrown at this call stack:
    [External Code]

Inner Exception 1:
TargetInvocationException: Exception has been thrown by the target of an invocation.

Inner Exception 2:
DllNotFoundException: Unable to load DLL 'LibTorchSharp' or one of its dependencies: The specified module could not be found. (0x8007007E)
```

**Comment by luisquintanilla** [comment]
When training on local GPU, it should reference Torchsharp-cuda-windows. When training on local CPU it should be Torchsharp-cpu

**Comment by luisquintanilla** [comment]
Let's also make sure it works for the Web API.

**Comment by zewditu** [comment]
I think this should be done from ml net , why `SweepableEstimator.NugetDependencies` does not give us it dependencies ?
@LittleLittleCloud  thought?
A temp fix could be done from model builder

**Comment by luisquintanilla** [comment]
@zewditu is it a large change in ML.NET?

**Comment by JakeRadMSFT** [comment]
https://github.com/dotnet/machinelearning/blob/61b1fa5cf2e336a933ebd28df38b13df3ee7491b/src/Microsoft.ML.AutoML/CodeGen/trainer-estimators.json#L527

This probably should have the libtorch dependency listed. We'd have to choose either GPU or CPU though. 

I think we can have a better experience in MB at the moment. 

@zewditu do you have a recommended approach on the MB side?

**Comment by zewditu** [comment]
> @zewditu is it a large change in ML.NET?

I am not sure , I did not know Ml.net repo very well ,  but my best guss is it is not large

**Comment by zewditu** [comment]
> https://github.com/dotnet/machinelearning/blob/61b1fa5cf2e336a933ebd28df38b13df3ee7491b/src/Microsoft.ML.AutoML/CodeGen/trainer-estimators.json#L527
> 
> This probably should have the libtorch dependency listed. We'd have to choose either GPU or CPU though.
> 
> I think we can have a better experience in MB at the moment.
> 
> @zewditu do you have a recommended approach on the MB side?

Yes depending on  the scenario and the environment (GPU  or CPU) we can add  dependency  in the code gen.

**Comment by zewditu** [comment]
The  fix in Ml net should be something like this PR https://github.com/dotnet/machinelearning/pull/6408

**Comment by luisquintanilla** [comment]
Validated. Fixed. Closing this issue.

**Comment by torronen** [comment]
I having an error which I think is relatd "This application or script uses TorchSharp but doesn't contain a reference to libtorch-cude-11.3.-win-x64, Version 1.11.0.1"

Installing libtorch-cuda-11.3.-win-x64 packages from nuget did not fix the issue.
Problem also remains after installing TorchSharp-cuda-windows (but I may have wrong version of it).

I was able to start the training once, but after that I have this error in that solution and also a new solution .NET 6.0

In the middle of the succesfully started and then the new one I updated to Nvidia RTX Studio drivers instead of game-ready earlier. I suppose it is not this, because it says about the reference, but I thought to mention it just in case.

![image](https://user-images.githubusercontent.com/26261427/201218739-e1968074-617b-43d3-865e-ef8b2ac9d436.png)
